hq/app/aws/support/TrustedAdvisorS3.scala

```diff
@@ -31,25 +31,26 @@ object TrustedAdvisorS3 {
 
 //  When you use server-side encryption, Amazon S3 encrypts an object before saving
 //  it to disk in its data centers and decrypts it when you download the object
-  private def addEncryptionStatus(bucket: BucketDetail, client: AwsClient[AmazonS3])(implicit ec: ExecutionContext): BucketDetail = {
+  private def addEncryptionStatus(bucket: BucketDetail, account: AwsAccount, clients: AwsClients[AmazonS3])(implicit ec: ExecutionContext): Attempt[BucketDetail] = {
     // If there is no bucket encryption, AWS returns an error...
     // Assume bucket is not encrypted if we cannot successfully getBucketEncryption
-    try {
-      client.client.getBucketEncryption(bucket.bucketName)
-      bucket.copy(isEncrypted = true)
-    } catch {
-        case e: AmazonS3Exception => {
-          bucket
-        }
+    val attemptFindEncryptionStatus = for {
+      region <- Try(Regions.fromName(bucket.region))
+      encryptionStatus <- Try(clients.get(account, region).map(_.client.getBucketEncryption(bucket.bucketName)))
+    } yield encryptionStatus
+
+    attemptFindEncryptionStatus match {
+      // If we fail to retrieve the correct client, then assume no encryption
+      case Success(attempt) => Attempt.Async.Right(attempt.fold(_ => bucket, _ => bucket.copy(isEncrypted = true)))
+      case scala.util.Failure(_) => Attempt.Right(bucket)
```

Comment:
This seems like a bit of a mis-use of attempt but I couldn't find a cleaner way of achieving the logic (wrapped in an attempt):
- If region is null or invalid, return `bucket`
- If client for the provided region is not found, return `bucket`
- If client is found and the request is unsuccessful return `bucket`
- Otherwise, when region, client, request all "good" return `bucket.copy(isEncrypted = true)`